File: test_supabase.py
import requests
import json

# Supabase configuration
SUPABASE_URL = 'https://ynkzcybctsstpqxdoweq.supabase.co'
SUPABASE_ANON_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Inlua3pjeWJjdHNzdHBxeGRvd2VxIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Njk4NzI1NjAsImV4cCI6MjA4NTQ0ODU2MH0.ZXdFcLmSmgaikZrA9MpP6d9enp4rjz_9nuiTpwm9n2k'

# Headers
headers = {
    'apikey': SUPABASE_ANON_KEY,
    'Authorization': f'Bearer {SUPABASE_ANON_KEY}',
    'Content-Type': 'application/json'
}

# Test 1: Check if we can connect to Supabase and get users
print("Testing Supabase connection...")

try:
    # Get all users
    response = requests.get(
        f'{SUPABASE_URL}/rest/v1/users?select=*',
        headers=headers
    )
    
    if response.status_code == 200:
        users = response.json()
        print(f"✅ Success! Found {len(users)} users")
        
        # Count users by role
        role_counts = {}
        for user in users:
            role = user.get('role', 'unknown')
            role_counts[role] = role_counts.get(role, 0) + 1
        
        print("\nUsers by role:")
        for role, count in role_counts.items():
            print(f"  {role}: {count}")
    else:
        print(f"❌ Error: {response.status_code} - {response.text}")

except Exception as e:
    print(f"❌ Error: {str(e)}")

# Test 2: Check if we can get modules
print("\n\nTesting modules table...")
try:
    response = requests.get(
        f'{SUPABASE_URL}/rest/v1/modules?select=*',
        headers=headers
    )
    
    if response.status_code == 200:
        modules = response.json()
        print(f"✅ Success! Found {len(modules)} modules")
    else:
        print(f"❌ Error: {response.status_code} - {response.text}")

except Exception as e:
    print(f"❌ Error: {str(e)}")

# Test 3: Check if we can get lessons{

File: test-supabase.js
// Test script to check Supabase connection and grade levels/sections
// Run this with: node test-supabase.js

const { createClient } = require('@supabase/supabase-js');

// Configuration
const SUPABASE_URL = 'https://ynkzcybctsstpqxdoweq.supabase.co';
const SUPABASE_ANON_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Inlua3pjeWJjdHNzdHBxeGRvd2VxIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Njk4NzI1NjAsImV4cCI6MjA4NTQ0ODU2MH0.ZXdFcLmSmgaikZrA9MpP6d9enp4rjz_9nuiTpwm9n2k';

// Initialize Supabase client
const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY);

async function testConnection() {
    console.log('Testing Supabase connection...');
    try {
        const { data, error } = await supabase
            .from('grade_levels')
            .select('*')
            .order('name');

        if (error) {
            console.error('❌ Error connecting to Supabase:', error);
            return false;
        }

        console.log(`✅ Connected successfully! Found ${data.length} grade levels.`);
        return true;
    } catch (error) {
        console.error('❌ Connection error:', error);
        return false;
    }
}

async function testSectionsQuery() {
    console.log('\nTesting sections query...');
    try {
        const { data, error } = await supabase
            .from('sections')
            .select(`
                *,
                grade_levels!sections_grade_level_id_fkey (
                    id,
                    name
                )
            `)
            .order('name');

        if (error) {
            console.error('❌ Error fetching sections:', error);
            return false;
        }

        console.log(`✅ Found ${data.length} sections.`);
        
        // Log section details
        data.forEach(section => {
            console.log(`  - ${section.name} (Grade: ${section.grade_levels?.name || 'N/A'})`);
        });

        return true;
    } catch (error) {
        console.error('❌ Sections query error:', error);
        return false;
    }
}

async function main() {
    console.log('=== MathTuro LMS - Grade Levels & Sections Test ===\n');

    const connected = await testConnection();
    if (!connected) {
        console.error('❌ Failed to connect to Supabase');
        process.exit(1);
    }

    const sectionsFound = await testSectionsQuery();
    if (!sectionsFound) {
        console.error('❌ Failed to fetch sections');
        process.exit(1);
    }

    console.log('\n✅ All tests passed!');
}

main().catch(error => {
    console.error('❌ Uncaught error:', error);
    process.exit(1);
});